File: 프로그래머스/1/12915. 문자열 내 마음대로 정렬하기/문자열 내 마음대로 정렬하기.java
import java.util.*;

class Solution {
    public String[] solution(String[] strings, int n) {
        List<String> list = Arrays.asList(strings);
        
        list.sort((a, b) -> {
            return a.charAt(n) == b.charAt(n) ? 
                a.compareTo(b) : a.charAt(n) - b.charAt(n);
        });
        String[] answer = list.toArray(new String[0]);
        return answer;
    }
}

File: 프로그래머스/1/12915. 문자열 내 마음대로 정렬하기/문자열 내 마음대로 정렬하기.cpp
#include <string>
#include <vector>
#include <map>
#include <set>

using namespace std;

vector<string> solution(vector<string> strings, int n) {
    vector<string> answer;
    map<char, set<string>> m;
    for(int i = 0; i < strings.size(); i++){
        m[strings[i][n]].insert(strings[i]);
    }

    for (const auto& pair : m) {
        for (const auto& s : pair.second) {
            answer.push_back(s);
        }
    }
    return answer;
}